django/db/backends/postgresql/schema.py

```diff
@@ -94,3 +94,41 @@ def _alter_column_type_sql(self, table, old_field, new_field, new_type):
             return super(DatabaseSchemaEditor, self)._alter_column_type_sql(
                 table, old_field, new_field, new_type
             )
+
+    def _alter_field(self, model, old_field, new_field, old_type, new_type,
+                     old_db_params, new_db_params, strict=False):
+        super(DatabaseSchemaEditor, self)._alter_field(
+            model, old_field, new_field, old_type, new_type, old_db_params,
+            new_db_params, strict
+        )
+        # Added an index?
+        if ((not old_field.db_index and new_field.db_index) or (not old_field.unique and new_field.unique)):
+            db_type = new_field.db_type(connection=self.connection)
+            if db_type is not None and (new_field.db_index or new_field.unique):
+                # Fields with database column types of `varchar` and `text` need
+                # a second index that specifies their operator class, which is
+                # needed when performing correct LIKE queries outside the
+                # C locale. See #12234.
+                #
+                # This code resambles the one found in `self._model_indexes_sql`.
+                # The only difference is that here the statements are
+                # executed immediately.
```

Comment:
I'm using `self.execute` because that is what the base class `_alter_field` method does. Honestly, I don't know why the statements are deferred during model creation and why are executed instantly during the alter.